File: Atcoder_#342/ruby/b_which_is_ahead?.rb
# 標準入力からデータを取得
n = gets.to_i
p = gets.split.map(&:to_i)
q = gets.to_i

# クエリを取得
queries = []
q.times do
  queries << gets.split.map(&:to_i)
end

# 各クエリに対して処理を実行
queries.each do |a, b|
  # それぞれの人が前から何番目かを調べる
  a_index = p.index(a)
  b_index = p.index(b)

  # より前に並んでいる人を出力
  if a_index < b_index
    puts a
  else
    puts b
  end
end

File: Atcoder_#342/javascript/b_which_is_ahead?.js
const readline = require('readline');

// 標準入力を読み取るためのインターフェースを作成
const rl = readline.createInterface({
  input: process.stdin,
  output: process.stdout
});

let inputLines = [];

// 標準入力を読み取るイベントリスナー
rl.on('line', (line) => {
  inputLines.push(line);
}).on('close', () => {
  // 入力データを処理する関数を呼び出し
  processInput(inputLines);
});

function processInput(input) {
  const n = parseInt(input[0]);
  const p = input[1].split(' ').map(Number);
  const q = parseInt(input[2]);
  const queries = input.slice(3).map(line => line.split(' ').map(Number));

  queries.forEach(([a, b]) => {
    // それぞれの人が前から何番目かを調べる
    const aIndex = p.indexOf(a);
    const bIndex = p.indexOf(b);

    // より前に並んでいる人を出力
    if (aIndex < bIndex) {
      console.log(a);
    } else {
      console.log(b);
    }
  });
}
